src/QMCDrivers/DMC/WalkerControl.cpp

```diff
@@ -234,13 +235,20 @@ int WalkerControl::branch(int iter, MCPopulation& pop, bool do_not_branch)
     }
   }
 
+  if (walkers.size() - untouched_walkers > 0)
   {
     ScopedTimer recomputing_timer(my_timers_[WC_recomputing]);
+
+    if (!twfs_shared_resource_)
+      twfs_shared_resource_ = std::make_unique<ResourceCollection>(pop.get_golden_twf().getResource());
+
     const size_t num_walkers = walkers.size();
     // recomputed received and duplicated walkers, the first untouched_walkers walkers doesn't need to be updated.
     auto p_list =
         convertUPtrToRefVectorSubset(pop.get_elec_particle_sets(), untouched_walkers, num_walkers - untouched_walkers);
     auto wf_list = convertUPtrToRefVectorSubset(pop.get_twfs(), untouched_walkers, num_walkers - untouched_walkers);
+
+    ResourceCollectionLock<TrialWaveFunction> resource_lock(*twfs_shared_resource_, wf_list[0]);
```

Comment:
I am now confused about whether twfs_shared_resource_ is expected to be persistent state or not.  Looks to me like it shouldn't be.  We are giving block scope access to the wavefunctions. If I want to clone that resource collection I also expect it to be in the gold twf but now there is a nullptr there.

Reply:
WalkerControl is outside any crowds and it need some recompute capability. For this reason, it owns a collection of resource copied from the golden_twf. The golden TWF should not be touched.